examples/grpc-kotlin/src/main/kotlin/example/armeria/grpc/kotlin/ArmeriaContext.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package example.armeria.grpc.kotlin
+
+import com.linecorp.armeria.common.RequestContext
+import com.linecorp.armeria.server.ServiceRequestContext
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.asCoroutineDispatcher
+
+/**
+ * A [CoroutineContext] that propagates an [ServiceRequestContext] to coroutines run using
+ * an Armeria context aware executor.
+ */
+internal object ArmeriaContext : CoroutineContext {
```

Comment:
I don't have high hopes for https://github.com/grpc/grpc-kotlin/issues/66 being solved for a long time.

How about instead of hacking relying on implementation detail like this, go ahead and use the wrapper function pattern so users implement stubs something like

```kotlin

class HelloServiceImpl {
  override suspend function hello = withArmeriaContext {
  }
}
```

Reply:
But... to use `withArmeriaContext { ... }`, we should change the default dispatcher to `Dispatchers.Unconfined`

Reply:
I am not sure which one is better. 🤔

Reply:
Yeah - the hack is way too hacky though. If it's short lived it's ok, but I don't think we should recommend it for a long time. Our pattern can be the two rules

1) Use Unconfined
2) Decorate all methods with `withArmeriaContext`